pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -792,101 +795,58 @@ func (h *upgradeHandler) storeVMState(upgrade *harvesterv1.Upgrade) error {
 	}
 	runningVMsJSON, err := json.Marshal(runningVMs)
 	if err != nil {
-		logrus.WithFields(logrus.Fields{
-			"namespace": upgrade.Namespace,
-			"name":      upgrade.Name,
-		}).WithError(err).Error("Failed to marshal PreUpgradeRunningVM list to json")
+		logrus.WithFields(logFields).WithError(err).Error("Failed to marshal PreUpgradeRunningVM list to json")
 		return err
 	}
 
-	secret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: harvesterSystemNamespace,
-			Name:      preUpgradeVMsSecretName,
-			OwnerReferences: []metav1.OwnerReference{
-				upgradeReference(upgrade),
-			},
-		},
-		Data: map[string][]byte{
-			"vms": runningVMsJSON,
-		},
-	}
-	return h.createOrUpdateSecret(secret)
+	upgrade.Annotations[preUpgradeVMsAnnotation] = string(runningVMsJSON)
```

Comment:
Is it possible, at this time point, that the annotation is still `nil`?

Reply:
yeah that might happen, I'll fix that, thanks.